### PR TITLE
Update docs for process::abort

### PR DESCRIPTION
**Pull request by mistodon**
Remove a typo and explain the relationship to `panic!`.

Part of #29370 

r? @steveklabnik

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -1124,7 +1124,14 @@ pub fn exit(code: i32) -> ! {
 ///
 /// Note that because this function never returns, and that it terminates the
 /// process, no destructors on the current stack or any other thread's stack
-/// will be run. If a clean shutdown is needed it is recommended to only call
+/// will be run.
+///
+/// This is in contrast to the default behaviour of [`panic!`] which unwinds
+/// the current thread's stack and calls all destructors.
+/// When `panic="abort"` is set, either as an argument to `rustc` or in a
+/// crate's Cargo.toml, [`panic!`] and `abort` are equivalent.
```
Comment (diwic) [inline src/libstd/process.rs]: Is this correct? With panic=abort, I assumed a settable panic handler and/or a backtrace would happen, whereas process::abort will not call a panic handler.

Comment (Pirh) [reply src/libstd/process.rs]: Just tested this and it looks like you're right - I'd missed that. I'll change the wording and mentioned that process::abort skips the panic handler.

**Comment by Pirh** [comment]
I've amended that description to mention the panic hook, I think this is correct now.

**Comment by diwic** [comment]
@Pirh 
Looks good to me! (I'm not someone who has any rights to ask bors to do anything, so you'll have to wait for someone else to review/approve as well.)

**Review by steveklabnik — APPROVED** [review]
Thanks so much!

**Pull request merged**